kolibri/plugins/coach/assets/src/views/plan/CreateExamPage/index.vue

```diff
@@ -359,18 +396,47 @@
       },
     },
     watch: {
+      $route(to, from) {
+        if (this.$route.params.topicId) {
+          ContentNodeResource.fetchCollection({
+            getParams: { parent: this.$route.params.topicId },
+          }).then(bookmarks => {
+            this.bookmarkNodes = bookmarks;
+          });
+        }
+      },
```

Comment:
I haven't inspected this in detail, but generally route handling is not performed in components using `watch: $route`. (`watch` in general is often a brittle tool). In most other places, routes populate vuex state which then drives component rendering. It's a bit more boilerplate but keeps things easier to reason about.

Reply:
It might be helpful to reference some of the other places where links are used in this file, like the [topicsLink](https://github.com/learningequality/kolibri/blob/b46ec1b5b0376dcd8fc86fb9628e32cb091042e5/kolibri/plugins/coach/assets/src/views/plan/CreateExamPage/index.vue#L575) as a starting point for getting familiar with our routing. It's taken me a while to wrap my head around how and where we are use route objects in Kolibri -- I still manage to get feedback about it every PR 😄 -- so maybe we can chat on Monday about this when I sign on!

Reply:
I changed the fetching logic to go into route handlers